iroh/src/rpc_protocol.rs

```diff
@@ -162,29 +167,40 @@ impl ServerStreamingMsg<ProviderService> for ListIncompleteBlobsRequest {
 ///
 /// Lists all collections that have been explicitly added to the database.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct ListCollectionsRequest;
+pub struct ListRootsRequest;
 
 /// A response to a list collections request
 #[derive(Debug, Serialize, Deserialize)]
-pub struct ListCollectionsResponse {
-    /// Hash of the collection
+pub struct ListRootsResponse {
+    /// Name of the root
+    pub name: Bytes,
+    /// Hash of the root
     pub hash: Hash,
-    /// Number of children in the collection
-    ///
-    /// This is an optional field, because the data is not always available.
-    pub total_blobs_count: Option<u64>,
-    /// Total size of the raw data referred to by all links
-    ///
-    /// This is an optional field, because the data is not always available.
-    pub total_blobs_size: Option<u64>,
+    /// Format of the root
+    pub format: Format,
 }
 
-impl Msg<ProviderService> for ListCollectionsRequest {
+impl Msg<ProviderService> for ListRootsRequest {
     type Pattern = ServerStreaming;
 }
 
-impl ServerStreamingMsg<ProviderService> for ListCollectionsRequest {
-    type Response = ListCollectionsResponse;
+impl ServerStreamingMsg<ProviderService> for ListRootsRequest {
+    type Response = ListRootsResponse;
+}
+
+/// List all collections
+///
+/// Lists all collections that have been explicitly added to the database.
```

Comment:
i know this is a WIP, but figured i'd just mention this comment hasn't been updated